File: FDK/FDK/SoftFX.Extended.Net/Exceptions/SendException.cs
namespace SoftFX.Extended.Errors
{
    using System;
    using System.Runtime.Serialization;
    using System.Security.Permissions;
    using SoftFX.Extended.Core;

    /// <summary>
    /// This exception indicates that outgoing request has not been sent.
    /// </summary>
    [Serializable]
    public class SendException : RuntimeException
    {
        internal SendException(string message)
            : base(HResults.FX_CODE_ERROR_SEND, message)
        {
        }

        /// <summary>
        /// 
        /// </summary>
        /// <param name="info"></param>
        /// <param name="context"></param>
        [SecurityPermissionAttribute(SecurityAction.Demand, SerializationFormatter = true)]
        protected SendException(SerializationInfo info, StreamingContext context)
            : base(info, context)
        {
        }
    }
}

File: FDK/Core/SendException.cpp
#include "stdafx.h"
#include "SendException.h"


CSendException::CSendException()
    : std::runtime_error("Server request has not been send.")
{
}

CSendException::CSendException(const char* message)
    : std::runtime_error(message)
{
}

CSendException::CSendException(const std::string& message)
    : std::runtime_error(message.c_str())
{
}
